# Подготовка файлов и программ

Некоторые программы ставятся дважды из разных источников для демонстрации различных вариантов в случае неполадок. Для оптимизации времени работы необходимо удалить лишний код.

In [2]:
pip install -q condacolab

In [3]:
import condacolab

In [4]:
condacolab.install()

⏬ Downloading https://github.com/jaimergp/miniforge/releases/latest/download/Mambaforge-colab-Linux-x86_64.sh...
📦 Installing...
📌 Adjusting configuration...
🩹 Patching environment...
⏲ Done in 0:00:35
🔁 Restarting kernel...


In [1]:
!conda install -c bioconda seqtk

Solving environment: \ | / - \ | / done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - seqtk


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge
    certifi-2021.10.8          |   py37h89c1867_1         145 KB  conda-forge
    conda-4.11.0               |   py37h89c1867_2        16.9 MB  conda-forge
    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda
    ------------------------------------------------------------
                                           Total:        17.2 MB

The following NEW packages will be INSTALLED:

  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3

The following packages will be UPDATED:

  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0
  certifi                      

In [2]:
!conda install -c bioconda trimmomatic

Solving environment: - \ | / - \ | / - \ | / done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - trimmomatic


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge
    python_abi-3.7             |          2_cp37m           4 KB  conda-forge
    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda
    ------------------------------------------------------------
                                           Total:        97.8 MB

The following NEW packages will be INSTALLED:

  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0
  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2

The following packages will be UPDATED:

  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m



trimmomatic-0.

In [3]:
!pip install macs2 intervene

     |████████████████████████████████| 2.1 MB 24.5 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
     |████████████████████████████████| 5.5 MB 50.8 MB/s 
     |████████████████████████████████| 12.5 MB 32.6 MB/s 
     |████████████████████████████████| 11.2 MB 65.3 MB/s 
     |████████████████████████████████| 11.3 MB 46.0 MB/s 
  Using cached numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
     |████████████████████████████████| 38.1 MB 1.3 MB/s 
     |████████████████████████████████| 292 kB 45.9 MB/s 
     |████████████████████████████████| 98 kB 7.1 MB/s 
     |████████████████████████████████| 247 kB 54.1 MB/s 
     |████████████████████████████████| 898 kB 65.9 MB/s 
     |████████████████████████████████| 1.1 MB 58.3 MB/s 
     |████████████████████████████████| 4.3 MB 39.0 MB/s 
     |████████████████████████████████| 40 kB 5.2 MB/s 
     |███████████████████

In [4]:
!apt-get install bedtools

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  bedtools
0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.
Need to get 577 kB of archives.
After this operation, 2,040 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]
Fetched 577 kB in 1s (967 kB/s)
Selecting previously unselected package bedtools.
(Reading database ... 155335 files and directories currently installed.)
Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...
Unpacking bedtools (2.26.0+dfsg-5) ...
Setting up bedtools (2.26.0+dfsg-5) ...


In [5]:
!apt-get install bowtie2 samtools

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc
  python-asn1crypto python-avro python-cachecontrol python-certifi
  python-cffi-backend python-chardet python-cryptography python-enum34
  python-html5lib python-idna python-ipaddress python-isodate python-lockfile
  python-mistune python-openssl python-pkg-resources python-pyparsing
  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml
  python-schema-salad python-shellescape python-six python-sparqlwrapper
  python-typing python-urllib3 python-webencodings
Suggested packages:
  bowtie2-examples python-cryptography-doc python-cryptography-vectors
  python-enum34-doc python-genshi python-lxml python-lockfile-doc
  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc
  python-rdflib-doc python-rdflib-tools python-socks python-ntlm
The f

In [6]:
!conda install -c bioconda bowtie2

Solving environment: - \ | / - \ | / - \ | done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - bowtie2


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    bowtie2-2.3.5.1            |   py37he513fc3_0        12.0 MB  bioconda
    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge
    tbb-2020.2                 |       h4bd325d_4         1.5 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        27.9 MB

The following NEW packages will be INSTALLED:

  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py37he513fc3_0
  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5
  tbb                conda-forge/linux-64::tbb-2020.2-h4bd325d_4



perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:05<00:00,  5.35s

## Установка FastQC

In [ ]:
!java -version

openjdk version "1.8.0_312"
OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)
OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)


In [1]:
!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip

--2022-03-10 20:19:55--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip
Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4
Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10249221 (9.8M) [application/zip]
Saving to: ‘fastqc_v0.11.9.zip’

fastqc_v0.11.9.zip   95%[==================> ]   9.30M  62.2KB/s    in 2m 25s  

2022-03-10 20:22:55 (65.7 KB/s) - Read error at byte 9751849/10249221 (Connection reset by peer). Retrying.

--2022-03-10 20:22:56--  (try: 2)  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip
Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 10249221 (9.8M), 497372 (486K) remaining [application/zip]
Saving to: ‘fast

In [7]:
!unzip fastqc_v0.11.9.zip

Archive:  fastqc_v0.11.9.zip
  inflating: FastQC/cisd-jhdf5.jar   
   creating: FastQC/Configuration/
  inflating: FastQC/Configuration/adapter_list.txt  
  inflating: FastQC/Configuration/contaminant_list.txt  
  inflating: FastQC/Configuration/limits.txt  
  inflating: FastQC/fastqc           
  inflating: FastQC/fastqc_icon.ico  
   creating: FastQC/Help/
   creating: FastQC/Help/1 Introduction/
   creating: FastQC/Help/1 Introduction/.svn/
  inflating: FastQC/Help/1 Introduction/.svn/entries  
   creating: FastQC/Help/1 Introduction/.svn/props/
   creating: FastQC/Help/1 Introduction/.svn/text-base/
  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  
   creating: FastQC/Help/1 Introduction/.svn/tmp/
   creating: FastQC/Help/1 Introduction/.svn/tmp/props/
  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  
   creating: FastQC/Help/2 Basic Operations/
   creating: FastQC/Help/2 Basic Operations/.svn/
  inflating: FastQC/Help/2 Basic

In [8]:
!chmod a+x FastQC/fastqc

In [9]:
!./FastQC/fastqc --help


            FastQC - A high throughput sequence QC analysis tool

SYNOPSIS

	fastqc seqfile1 seqfile2 .. seqfileN

    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] 
           [-c contaminant file] seqfile1 .. seqfileN

DESCRIPTION

    FastQC reads a set of sequence files and produces from each one a quality
    control report consisting of a number of different modules, each one of 
    which will help to identify a different potential type of problem in your
    data.
    
    If no files to process are specified on the command line then the program
    will start as an interactive graphical application.  If files are provided
    on the command line then the program will run with no user interaction
    required.  In this mode it is suitable for inclusion into a standardised
    analysis pipeline.
    
    The options for the program as as follows:
    
    -h --help       Print this help file and exit
    
    -v --version    Print the version of the program and exit

# Выравнивание

## Скачиваем чтения

Для примера чтения были взяты из эксперимента https://www.encodeproject.org/experiments/ENCSR000AKC/

In [10]:
# ChIP-seq на гистоновой метке
!wget https://www.encodeproject.org/files/ENCFF019SLS/@@download/ENCFF019SLS.fastq.gz
!wget https://www.encodeproject.org/files/ENCFF641KVN/@@download/ENCFF641KVN.fastq.gz
# ChIP-seq контроль
!wget https://www.encodeproject.org/files/ENCFF411GHV/@@download/ENCFF411GHV.fastq.gz


--2022-03-10 20:29:11--  https://www.encodeproject.org/files/ENCFF019SLS/@@download/ENCFF019SLS.fastq.gz
Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144
Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.
HTTP request sent, awaiting response... 307 Temporary Redirect
Location: https://encode-public.s3.amazonaws.com/2019/04/27/056e7433-bb81-4591-8271-57fdb881ba89/ENCFF019SLS.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF019SLS.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=yK5b%2FsC7LuTuM7ME7TvMdnfPhgw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJ

In [11]:
# распаковка ахрива
!gzip -d /content/ENCFF641KVN.fastq.gz /content/ENCFF411GHV.fastq.gz /content/ENCFF019SLS.fastq.gz

## FastQC (или multiQC может быть использован в работе)

In [12]:
!./FastQC/fastqc /content/ENCFF641KVN.fastq
!./FastQC/fastqc /content/ENCFF411GHV.fastq
!./FastQC/fastqc /content/ENCFF019SLS.fastq

Started analysis of ENCFF641KVN.fastq
Approx 5% complete for ENCFF641KVN.fastq
Approx 10% complete for ENCFF641KVN.fastq
Approx 15% complete for ENCFF641KVN.fastq
Approx 20% complete for ENCFF641KVN.fastq
Approx 25% complete for ENCFF641KVN.fastq
Approx 30% complete for ENCFF641KVN.fastq
Approx 35% complete for ENCFF641KVN.fastq
Approx 40% complete for ENCFF641KVN.fastq
Approx 45% complete for ENCFF641KVN.fastq
Approx 50% complete for ENCFF641KVN.fastq
Approx 55% complete for ENCFF641KVN.fastq
Approx 60% complete for ENCFF641KVN.fastq
Approx 65% complete for ENCFF641KVN.fastq
Approx 70% complete for ENCFF641KVN.fastq
Approx 75% complete for ENCFF641KVN.fastq
Approx 80% complete for ENCFF641KVN.fastq
Approx 85% complete for ENCFF641KVN.fastq
Approx 90% complete for ENCFF641KVN.fastq
Approx 95% complete for ENCFF641KVN.fastq
Analysis complete for ENCFF641KVN.fastq
Started analysis of ENCFF411GHV.fastq
Approx 5% complete for ENCFF411GHV.fastq
Approx 10% complete for ENCFF411GHV.fastq
Appr

## Выравнивание на хромосому

Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.

Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/

In [13]:
!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr18.fa.gz
!gzip -d chr18.fa.gz

--2022-03-10 20:32:04--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr18.fa.gz
Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163
Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 25154367 (24M) [application/x-gzip]
Saving to: ‘chr18.fa.gz’

chr18.fa.gz         100%[===================>]  23.99M  18.9MB/s    in 1.3s    

2022-03-10 20:32:06 (18.9 MB/s) - ‘chr18.fa.gz’ saved [25154367/25154367]



In [14]:
%%time
!bowtie2-build chr18.fa chromosome_index

Settings:
  Output files: "chromosome_index.*.bt2"
  Line rate: 6 (line is 64 bytes)
  Lines per side: 1 (side is 64 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  chr18.fa
Building a SMALL index
Reading reference sizes
  Time reading reference sizes: 00:00:01
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
bmax according to bmaxDivN setting: 20022401
Using parameters --bmax 15016801 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 15016801 --dcv 1024
Constructing suffix-arr

In [15]:
!mkdir bowtie2_res

In [16]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF019SLS.fastq \
          -S bowtie2_res/res_SLS.sam

4697829 reads; of these:
  4697829 (100.00%) were unpaired; of these:
    4478193 (95.32%) aligned 0 times
    100272 (2.13%) aligned exactly 1 time
    119364 (2.54%) aligned >1 times
4.68% overall alignment rate
CPU times: user 1.46 s, sys: 196 ms, total: 1.65 s
Wall time: 3min 13s


In [17]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF641KVN.fastq \
          -S bowtie2_res/res_KVN.sam

6467900 reads; of these:
  6467900 (100.00%) were unpaired; of these:
    6181873 (95.58%) aligned 0 times
    140146 (2.17%) aligned exactly 1 time
    145881 (2.26%) aligned >1 times
4.42% overall alignment rate
CPU times: user 2.03 s, sys: 273 ms, total: 2.3 s
Wall time: 4min 19s


In [18]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF411GHV.fastq \
          -S bowtie2_res/res_GHV.sam

2349952 reads; of these:
  2349952 (100.00%) were unpaired; of these:
    2106816 (89.65%) aligned 0 times
    80393 (3.42%) aligned exactly 1 time
    162743 (6.93%) aligned >1 times
10.35% overall alignment rate
CPU times: user 1.05 s, sys: 137 ms, total: 1.18 s
Wall time: 2min 14s


Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?

Имеет смысл для дальнейшего анализа отобрать уникально картированные риды.

In [19]:
import pandas as pd
data = {'Всего ридов':[4697829, 6467900, 2349952],
        'Выровнилось уникально ':[100272, 140146, 80393],
        'Выровнилось неуникально':[119364, 145881, 162743],
        'Не выровнилось':[4478193, 6181873, 2106816]}
 
frame = pd.DataFrame(data, index =['ENCFF019SLS',
                                'ENCFF641KVN',
                                'ENCFF411GHV'])
frame

Всего ридов  Выровнилось уникально   Выровнилось неуникально  \
ENCFF019SLS      4697829                  100272                   119364   
ENCFF641KVN      6467900                  140146                   145881   
ENCFF411GHV      2349952                   80393                   162743   

             Не выровнилось  
ENCFF019SLS         4478193  
ENCFF641KVN         6181873  
ENCFF411GHV         2106816

Процент выравниваний получился низким, из за того что выравнивание ридов производилось на одну так какхромосому

Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)

In [ ]:
# !samtools view -h -S -b -o bowtie2_res/res_ASU.bam bowtie2_res/res_ASU.sam
# !samtools view -h -S -b -o bowtie2_res/res_ARK.bam bowtie2_res/res_ARK.sam

На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя.

## Peak calling

Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3.

In [20]:
!mkdir macs2

In [21]:
!macs2 callpeak --broad -t bowtie2_res/res_SLS.sam \
    -c bowtie2_res/res_GHV.sam \
 	  -f SAM \
	  --outdir macs2 \
	  -n ENCFF019SLS

INFO  @ Thu, 10 Mar 2022 20:44:23: 
# Command line: callpeak --broad -t bowtie2_res/res_SLS.sam -c bowtie2_res/res_GHV.sam -f SAM --outdir macs2 -n ENCFF019SLS
# ARGUMENTS LIST:
# name = ENCFF019SLS
# format = SAM
# ChIP-seq file = ['bowtie2_res/res_SLS.sam']
# control file = ['bowtie2_res/res_GHV.sam']
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is on
# Paired-End mode is off
 
INFO  @ Thu, 10 Mar 2022 20:44:23: #1 read tag files... 
INFO  @ Thu, 10 Mar 2022 20:44:23: #1 read treatment tags... 
INFO  @ Thu, 10 Mar 2022 20:44:27: #1.2 read input 

In [22]:
!macs2 callpeak --broad -t bowtie2_res/res_KVN.sam \
    -c bowtie2_res/res_GHV.sam \
 	  -f SAM \
	  --outdir macs2 \
	  -n ENCFF641KVN

INFO  @ Thu, 10 Mar 2022 20:44:34: 
# Command line: callpeak --broad -t bowtie2_res/res_KVN.sam -c bowtie2_res/res_GHV.sam -f SAM --outdir macs2 -n ENCFF641KVN
# ARGUMENTS LIST:
# name = ENCFF641KVN
# format = SAM
# ChIP-seq file = ['bowtie2_res/res_KVN.sam']
# control file = ['bowtie2_res/res_GHV.sam']
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is on
# Paired-End mode is off
 
INFO  @ Thu, 10 Mar 2022 20:44:34: #1 read tag files... 
INFO  @ Thu, 10 Mar 2022 20:44:34: #1 read treatment tags... 
INFO  @ Thu, 10 Mar 2022 20:44:40: #1.2 read input 

In [23]:
!wget https://www.encodeproject.org/files/ENCFF546HBF/@@download/ENCFF546HBF.bed.gz
!gzip -d ENCFF546HBF.bed.gz

--2022-03-10 20:44:48--  https://www.encodeproject.org/files/ENCFF546HBF/@@download/ENCFF546HBF.bed.gz
Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144
Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.
HTTP request sent, awaiting response... 307 Temporary Redirect
Location: https://encode-public.s3.amazonaws.com/2021/05/05/9b56f0b0-0a2b-4169-a839-67332bc4fa35/ENCFF546HBF.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF546HBF.bed.gz&AWSAccessKeyId=ASIATGZNGCNXZWYIVGBT&Signature=bepas7KSYdFIuwvP5mAOPjov0VU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEIz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJIMEYCIQDaY8r2KIMQ8ARjUrel4Z6jykxbuFMGVcD0w7VvEM1OiQIhAPKUaxlYciwOyweAaoEYFpKNivlgI1vuXAg8hxwiTzZmKoMECPX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMjIwNzQ4NzE0ODYzIgzBB7D2VncA0vaa8%2FQq1wPCjkTyUMOR5eVVmBJuyCcR0UYJcNEXpNK2e%2BJV9t9bUw6BryIvV7Yfn28SsUnpHlFQbvU7BSuPEkRQHuk3%2B%2BppyOMX%2BESx5EKkqOnRwBdSrQscAGJoxgkYSXg

## Сравнение результатов

Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.

Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?

In [24]:
!intervene venn -i macs2/ENCFF019SLS_peaks.broadPeak ENCFF546HBF.bed  --filenames 


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ /content/Intervene_results. 
Thank you for using Intervene!



In [25]:
!intervene venn -i ENCFF546HBF.bed macs2/ENCFF019SLS_peaks.broadPeak  --filenames


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ /content/Intervene_results. 
Thank you for using Intervene!



In [26]:
!intervene venn -i macs2/ENCFF641KVN_peaks.broadPeak ENCFF546HBF.bed  --filenames


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ /content/Intervene_results. 
Thank you for using Intervene!



In [27]:
!intervene venn -i ENCFF546HBF.bed macs2/ENCFF641KVN_peaks.broadPeak  --filenames


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ /content/Intervene_results. 
Thank you for using Intervene!

